src/main/java/org/opensearch/neuralsearch/processor/TextImageEmbeddingProcessor.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.env.Environment;
+import org.opensearch.index.mapper.MapperService;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.neuralsearch.ml.MLCommonsClientAccessor;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * This processor is used for user input data text and image embedding processing, model_id can be used to indicate which model user use,
+ * and field_map can be used to indicate which fields needs embedding and the corresponding keys for the embedding results.
+ */
+@Log4j2
+public class TextImageEmbeddingProcessor extends AbstractProcessor {
```

Comment:
Can we extend NLPProcessor?

Reply:
I was checking that, but now NLPProcessor is restricted to a certain input and output format that are stick to a text embeddings. We should refactor that to a more generic code so other processors can re-use it, but per my estimations it goes beyond the 2.11 release timeline.